tests/unit/util/hse_err_test.c

```diff
@@ -57,24 +60,24 @@ MTF_DEFINE_UTEST(hse_err_test, merr_test_1)
 
     (void)merr_strinfo(err, errinfo, sizeof(errinfo), 0);
     errmsg = strerror_r(merr_errno(err), errbuf, sizeof(errbuf));
-    ASSERT_EQ(0, strcmp(strstr(errinfo, "I"), errmsg));
+    ASSERT_EQ(0, strncmp(strstr(errinfo, "I"), errmsg, strlen(errmsg)));
 
     file = hse_merr_base;
     err = merr_pack(EAGAIN, 0, file, 456);
     ASSERT_EQ(EAGAIN, merr_errno(err));
     ASSERT_EQ(456, merr_lineno(err));
     ASSERT_EQ(NULL, merr_file(err));
 
-    err = merr_pack(EBUG, 0, (char *)1, 123);
-    ASSERT_EQ(EBUG, merr_errno(err));
-    ASSERT_EQ(123, merr_lineno(err));
-    ASSERT_EQ(0, strcmp(merr_file(err), hse_merr_bug0));
-
-    file = hse_merr_base + sizeof(file);
+    file = (char*)&__start_hse_merr - MERR_ALIGN;
     err = merr_pack(EAGAIN, 0, file, 456);
     ASSERT_EQ(EAGAIN, merr_errno(err));
     ASSERT_EQ(456, merr_lineno(err));
-    ASSERT_EQ(0, strcmp(merr_file(err), hse_merr_bug1));
+    ASSERT_EQ(0, strcmp(merr_file(err), invalid_file_bug0));
```

Comment:
Please add a comment before line 71 to say that we testing a file ptr that is validly aligned but not within the hse_merr section.